### ISSUE TITLE
Cannot read property 'contents' of undefined

### ISSUE THREAD

**Issue by CatsMiaow** [TimeForANinja/node-ytsr#103]
- Node.js v14
- ytsr v3.2.3

### Case1
```sh
TypeError: Cannot read property 'contents' of undefined
    at module.exports (/home/bot/node_modules/ytsr/lib/main.js:54:28)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```
https://github.com/TimeForANinja/node-ytsr/blob/7a74b72aaf0da06c0d938bf6bb98eea0e008e744/lib/main.js#L54

Code used
```ts
await ytsr('word', { limit: 10 });
```

### Case2
```sh
TypeError: Cannot read property 'subMenu' of undefined
    at Object.exports.parseFilters (/home/bot/node_modules/ytsr/lib/utils.js:21:33)
    at AsyncFunction.main.getFilters (/home/bot/node_modules/ytsr/lib/main.js:143:15)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```
https://github.com/TimeForANinja/node-ytsr/blob/7a74b72aaf0da06c0d938bf6bb98eea0e008e744/lib/utils.js#L21

Code used
```ts
await ytsr.getFilters('word');
```

It doesn't seem to be a specific word problem, and it's occasionally happening.

**Comment by TimeForANinja** [comment]
🤔 the latest build just broke for a similar reason
https://github.com/TimeForANinja/node-ytsr/runs/1790452586

**Comment by CatsMiaow** [comment]
### Case3
```sh
TypeError: Cannot read property 'itemSectionRenderer' of undefined
    at parsePage2 (/home/bot/node_modules/ytsr/lib/main.js:97:82)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at module.exports (/home/bot/node_modules/ytsr/lib/main.js:78:22)
```

### Case4
```sh
Error: type richShelfRenderer is not known
    at parseItem (/home/bot/node_modules/ytsr/lib/parseItem.js:69:13)
    at catchAndLogFunc (/home/bot/node_modules/ytsr/lib/parseItem.js:76:12)
    at module.exports (/home/bot/node_modules/ytsr/lib/parseItem.js:99:46)
    at /home/bot/node_modules/ytsr/lib/main.js:54:34
    at Array.map (<anonymous>)
    at module.exports (/home/bot/node_modules/ytsr/lib/main.js:54:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

/***************************************************************************************************************
failed at func parseItem: type richShelfRenderer is not known
pls post the the files in /home/bot/node_modules/ytsr/dumps to https://github.com/TimeForANinja/node-ytsr/issues
os: linux-x64, node.js: v14.15.4, ytpl: 3.2.4
***************************************************************************************************************\
```
dumps [80pe2o4b4s-1612373458066.txt](https://github.com/TimeForANinja/node-ytsr/files/5922710/80pe2o4b4s-1612373458066.txt)

**Comment by TimeForANinja** [comment]
# Case 1 and 2
https://github.com/TimeForANinja/node-ytsr/commit/4de702aa223babb39f5f729adf9465eda08b51db
available on git & npm as v3.2.4


# Case 3 and Case 4
fixed in https://github.com/TimeForANinja/node-ytsr/commit/8c5dc75e14a84d536fbd6d55855c6afec45e6dc4
available on git & npm as v3.3.0

thanks again for reporting and providing dumps 👍
